lnrpc/walletrpc/walletkit_server.go

```diff
@@ -1448,6 +1452,25 @@ func (w *WalletKit) ListAccounts(ctx context.Context,
 	return &ListAccountsResponse{Accounts: rpcAccounts}, nil
 }
 
+// RequiredReserve returns the required reserve for a wallet.
+func (w *WalletKit) RequiredReserve(ctx context.Context,
+	req *RequiredReserveRequest) (*RequiredReserveResponse, error) {
+
+	numAnchorChans, err := w.cfg.CurrentNumAnchorChans()
+	if err != nil {
+		return nil, err
+	}
+
+	anchorChanReservedValue := lnwallet.AnchorChanReservedValue
```

Comment:
If you re-order the commits and add the wallet balance changes first, you could simply call `w.cfg.Wallet.RequiredReserve(totalChans)` here.